### ISSUE TITLE
parent and dirrename should return In MonadThrow m

### ISSUE THREAD

**Issue by harendra-kumar** [commercialhaskell/path#33]
Should this be fixed?

```
>> isParentOf $(mkAbsDir "/") $(mkAbsDir "/")
False
>> parent $(mkAbsDir "/")
"/"
>> 
```

**Comment by chrisdone** [comment]
It's definitely an inconsistency.

**Comment by harendra-kumar** [comment]
There are two ways to fix that. Let `parent` throw an exception on `/` or let `isParentOf` return true in case of `/`. Unix traditionally allows `cd ..` when you are in root dir so the latter might be in line with that.

**Comment by chrisdone** [comment]
Indeed, the magicness of the root directory seems to give us no option but to pick something not great. `isParentOf` returning true for the same two arguments seems to mess with people's intuition of hierarchy.

Putting `parent` in `MonadThrow m` is inconvenient, but it is more correct. Perhaps we can plan a migration path.

**Comment by harendra-kumar** [comment]
The two options wrt root dir sound like this:
1) Sorry, its a dead end you can't go beyond this 
2) You try to go up, you succeed and you find yourself in the same place

The first one is a more explicit and intuitive handling than the latter. Looks like a better option to me as well though a bit more inconvenient compared to the other one.

**Comment by chrisdone** [comment]
Indeed. I think (1) is explicit and the kind of thing this library is about.

**Comment by harendra-kumar** [comment]
Throw an exception or return a `Maybe`?

**Comment by mrkkrp** [comment]
I think it should throw just like all the other functions. It's more flexible and you can always get a `Maybe` value from `MonadThrow m => ... m a`.

**Comment by mrkkrp** [comment]
Should this be considered a duplicate of #18? Can we close it in favor of that issue?

**Comment by mrkkrp** [comment]
Closing in favor of #18. Should be resolved in version 0.6.0
